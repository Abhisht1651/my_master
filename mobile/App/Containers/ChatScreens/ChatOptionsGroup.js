/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable no-shadow */
// @flow
import { NavigationContainerProps } from '@react-navigation/native'
import React, { useState } from 'react'
import { View, Text, Image, FlatList, TouchableOpacity } from 'react-native'
import { createMaterialTopTabNavigator } from '@react-navigation/material-top-tabs'
import styles from './ScreenStyles/ChatOptionsGroupStyles'
import { Images } from '../../Themes'
import Elipsis from '../../Images/Icons/icon-elipsis.svg'
import { X2SButton } from '../../Components/Buttons/X2SButton'
import ListofUser from '../../Components/ListOfUsers/ListOfUsers'
import StorylineTab from '../../Components/ChatScreenComponents/StorylineTab'
import PollTab from '../../Components/ChatScreenComponents/PollTab'
import MediaTab from '../../Components/ChatScreenComponents/MediaTab'
import TabBar from '../../Components/ChatScreenComponents/TabBar'
import RenameGroup from './RenameGroup'

type Props = {
  item: {
    id: string,
    grpicon: Image,
    members: Array<{ id: string, name: string, image: Image, activeStatus: string }>,
    image: Image,
    name: string,
    activeStatus: string
  },
  route: {
    params: {
      title: string
    }
  }
} & NavigationContainerProps

const Tab = createMaterialTopTabNavigator()
const Data = [
  {
    id: '1'
  },
  {
    id: '2',
    members: [
      { id: '5', name: 'Tony', image: Images.voterIcon2, activeStatus: 'Active 45 minutes' },
      { id: '6', name: 'Steve', image: Images.voterIcon2, activeStatus: 'Active 45 minutes' },
      { id: '7', name: 'Banner', image: Images.voterIcon2, activeStatus: 'Active 45 minutes' },
      { id: '8', name: 'Natasha', image: Images.voterIcon2, activeStatus: 'Active 45 minutes' },
      { id: '9', name: 'HawkEye', image: Images.voterIcon2, activeStatus: 'Active 45 minutes' }
    ],
    groupname: '',
    grpicon: Images.profilePicture
  },
  {
    id: '3'
  }
]
const ChatOptionsHeader = (props: Props) => {
  const { title } = props.route.params
  const { navigation } = props

  // Rename Group
  const [RenamePopUpVisible, setRenamePopUpVisible] = useState(false)
  const openRenamePopUp = () => {
    setRenamePopUpVisible(true)
    console.log('Rename PopUp running')
  }
  const closeRenamePopUp = () => {
    setRenamePopUpVisible(false)
    console.log('Rename PopUp Close')
  }
  const [typedText, setTypedText] = useState('Rename group')
  const inputHandler = (text) => {
    setTypedText(text)
    console.log(typedText)
  }
  const changeGroupName = () => {
    setgrpname(typedText)
    console.log('Saved!')
    closeRenamePopUp()
  }
  const [grpname, setgrpname] = useState(title)
  const renderuser = ({ item }: Props) => {
    return (
      <View style={styles.listofuser}>
        <ListofUser
          name={item.name}
          image={item.image}
          optionalText={item.activeStatus}
          buttonText="Message"
        />
      </View>
    )
  }
  const renderitem = ({ item }: Props) => {
    if (item.id === '1') {
      return (
        <View style={styles.header}>
          <View style={styles.backbutton}>
            <TouchableOpacity onPress={() => navigation.navigate('Chat2', { title: grpname })}>
              <Image source={Images.arrowleft} style={styles.arrowleft} />
            </TouchableOpacity>
          </View>
          <Text style={styles.heading}>Chat options</Text>
          <View style={styles.elipsis}>
            <TouchableOpacity>
              <Elipsis />
            </TouchableOpacity>
          </View>
        </View>
      )
    }
    if (item.id === '2') {
      return (
        <View style={styles.container1}>
          <Image source={item.grpicon} style={styles.grpicon} />
          <View style={styles.cameracont}>
            <TouchableOpacity>
              <Image source={Images.camera2x} style={styles.cameraimg} />
            </TouchableOpacity>
          </View>
          {grpname === '' ? (
            [
              item.members.length === 2 ? (
                <Text style={styles.grpnametext}>
                  {item.members[0].name}, {item.members[1].name}
                </Text>
              ) : (
                <Text style={styles.grpnametext}>
                  {item.members[0].name}, {item.members[1].name}, +{item.members.length - 2}
                </Text>
              )
            ]
          ) : (
            <Text style={styles.grpnametext}>{grpname}</Text>
          )}
          <View style={styles.button}>
            <TouchableOpacity>
              <X2SButton title="Rename group" outline onPress={openRenamePopUp} />
            </TouchableOpacity>
            <RenameGroup
              isModalVisible={RenamePopUpVisible}
              onClose={closeRenamePopUp}
              captureText={inputHandler}
              onBtnPress={changeGroupName}
            />
          </View>
          <View>
            <Text style={styles.membertxt}>Members</Text>
            <View style={styles.addcont}>
              <TouchableOpacity>
                <Text style={styles.addtext}>Add more</Text>
              </TouchableOpacity>
            </View>
          </View>
          <View style={styles.flatlist}>
            <FlatList
              data={item.members}
              keyExtractor={(post) => post.id}
              renderItem={renderuser}
            />
          </View>
        </View>
      )
    }
    return (
      <Tab.Navigator tabBar={(props) => <TabBar {...props} />}>
        <Tab.Screen name="Storyline" component={StorylineTab} />
        <Tab.Screen name="Polls" component={PollTab} />
        <Tab.Screen name="Media" component={MediaTab} />
      </Tab.Navigator>
    )
  }

  return (
    <View style={styles.container}>
      <FlatList
        data={Data}
        keyExtractor={(item) => item.id}
        renderItem={renderitem}
        stickyHeaderIndices={[0]}
        listKey={(item) => `${item.id}D`}
      />
    </View>
  )
}

export default ChatOptionsHeader
