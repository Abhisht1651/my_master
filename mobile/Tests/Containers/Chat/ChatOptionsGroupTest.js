import React from 'react'
import { shallow } from 'enzyme'
import { FlatList, TouchableOpacity } from 'react-native'
import ChatOptionsGroup from '../../../App/Containers/ChatScreens/ChatOptionsGroup'

describe('ChatOptions', () => {
  it('ChatOptions component renders correctly', () => {
    const wrapper = shallow(<ChatOptionsGroup route={{ params: { title: ' ' } }} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render 1 flatlist', () => {
    const wrapper = shallow(<ChatOptionsGroup route={{ params: { title: ' ' } }} />)
    expect(wrapper.find(FlatList)).toHaveLength(1)
  })
  it('should render 2 TouchableOpacity', () => {
    const item = {
      id: '1'
    }
    const wrapper = shallow(<ChatOptionsGroup route={{ params: { title: ' ' } }} />)
    const flatlist = wrapper.find(FlatList)
    const element = flatlist.renderProp('renderItem')({ item })
    expect(element.find(TouchableOpacity)).toHaveLength(2)
  })
})
