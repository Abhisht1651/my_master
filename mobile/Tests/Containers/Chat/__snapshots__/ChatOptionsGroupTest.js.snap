// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatOptions ChatOptions component renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <FlatList
    data={
      Array [
        Object {
          "id": "1",
        },
        Object {
          "groupname": "",
          "grpicon": Object {
            "default": "",
          },
          "id": "2",
          "members": Array [
            Object {
              "activeStatus": "Active 45 minutes",
              "id": "5",
              "image": Object {
                "default": "",
              },
              "name": "Tony",
            },
            Object {
              "activeStatus": "Active 45 minutes",
              "id": "6",
              "image": Object {
                "default": "",
              },
              "name": "Steve",
            },
            Object {
              "activeStatus": "Active 45 minutes",
              "id": "7",
              "image": Object {
                "default": "",
              },
              "name": "Banner",
            },
            Object {
              "activeStatus": "Active 45 minutes",
              "id": "8",
              "image": Object {
                "default": "",
              },
              "name": "Natasha",
            },
            Object {
              "activeStatus": "Active 45 minutes",
              "id": "9",
              "image": Object {
                "default": "",
              },
              "name": "HawkEye",
            },
          ],
        },
        Object {
          "id": "3",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    listKey={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={
      Array [
        0,
      ]
    }
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;
